Bricks/exception.h

```diff
@@ -38,20 +39,45 @@ class Exception : public std::exception {
   Exception(const std::string& what) : what_(what) {}
   virtual ~Exception() = default;
 
-  void SetWhat(const std::string& what) { what_ = what; }
+  void SetWhat(const std::string& what) {
+    what_ = what;
+  }
 
+  virtual std::string DetailedDescription() const noexcept {
+    return strings::Printf("%s:%d", file_, line_) + '\t' + caller_ + '\t' + what_;
+  }
   // LCOV_EXCL_START
-  virtual const char* what() const noexcept override { return what_.c_str(); }
+  virtual const char* what() const noexcept override {
+    constexpr size_t MAX_LENGTH = 1024 * 10;
+    static char data[MAX_LENGTH + 1];
+    const std::string message = DetailedDescription();
+    if (message.length() < MAX_LENGTH) {
+      std::strcpy(data, message.data());
+    } else {
+      std::copy(message.begin(), message.begin() + MAX_LENGTH, data);
+      data[MAX_LENGTH] = '\0';
+    }
+    return data;
+  }
```

Comment:
So, `what()` and `What()` now return different things :-/
IMHO it becomes really not obvious, how to use this combination without consulting with the source code every time one needs to log an exception.